### PR TITLE
add upstream api

### PR DESCRIPTION
**Pull request by gxthrj**

### CODE REVIEW

**Comment by membphis** [comment]
is there no test cases? or we can add them later

**Comment by gxthrj** [comment]
> is there no test cases? or we can add them later

I will add some later

**Comment by membphis** [comment]
we'd better create an issue about this.

File: api/route/upstream.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	r.GET("/apisix/admin/upstreams", listUpstream)
+	r.PUT("/apisix/admin/upstreams/:uid", updateUpstream)
+	r.DELETE("/apisix/admin/upstreams/:uid", deleteUpstream)
+	return r
+}
+
+func createUpstream(c *gin.Context) {
+	u4 := uuid.NewV4()
+	uid := u4.String()
+	// todo 参数校验
```
Comment (moonming) [inline api/route/upstream.go:27]: English only please

**Pull request merged**